### PR TITLE
e2e & misc fixes for EnvoyPatchPolicy

### PR DESCRIPTION
**Pull request by arkodg**
misc fixes:
* Only add EnvoyPatchPolicy to provider resources if enabled in
  EnvoyGateway API
* Add `omitempty` tag to `priority` field to make it optional

e2e
* Use LocalReplyConfig to return a custom status code `406` when there is no valid route match

Relates to https://github.com/envoyproxy/gateway/issues/24

### CODE REVIEW

File: api/v1alpha1/envoypatchpolicy_types.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ type EnvoyPatchPolicySpec struct {
 	// the priority i.e. int32.min has the highest priority and
 	// int32.max has the lowest priority.
 	// Defaults to 0.
-	Priority int32 `json:"priority"`
+	Priority int32 `json:"priority,omitempty"`
```
Comment (zirain) [inline api/v1alpha1/envoypatchpolicy_types.go:62]: this's OK to me, but I recall the api guide said should be a pointer?

File: examples/redis/redis.yaml [hunk 2]
```diff
@@ -62,6 +62,8 @@ data:
       type: Kubernetes
     gateway:
       controllerName: gateway.envoyproxy.io/gatewayclass-controller
+    extensionApis:
```
Comment (zirain) [inline examples/redis/redis.yaml:65]: this seems not related to an `redis`example, can you address some comment?

File: api/v1alpha1/envoypatchpolicy_types.go (hunk 1 above)
Comment (arkodg) [reply api/v1alpha1/envoypatchpolicy_types.go:62]: its still required internally, `omitempty` allows the user to skip setting it when creating the resource and use the default value / `0` instead

File: examples/redis/redis.yaml (hunk 2 above)
Comment (arkodg) [reply examples/redis/redis.yaml:65]: this is the yaml used in e2e to update the startup config, I can start renaming and breaking things in separate files, would you prefer that ?

Comment (zirain) [reply examples/redis/redis.yaml:65]: yes, we can do that later

**Review by zirain — APPROVED** [review]

**Review by qicz — APPROVED** [review]
